graphqlapi/src/main/java/io/stargate/graphql/schema/NameConversions.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.graphql.schema;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+
+public class NameConversions {
+
+  enum IdentifierType {
+    TABLE,
+    UDT,
+    COLUMN,
+  }
+
+  private NameConversions() {}
+
+  public static String toGraphql(String cqlName, IdentifierType type) {
+    if (cqlName == null || cqlName.isEmpty()) {
+      throw new IllegalArgumentException("CQL name must be non-null and not empty");
+    }
+    String graphqlName;
+    if (type == IdentifierType.TABLE) {
+      graphqlName = RESERVED_GRAPHQL_NAMES.get(cqlName);
+      if (graphqlName != null) {
+        return graphqlName;
+      }
+    }
+    graphqlName = maybeHexEscape(cqlName);
+    if (type == IdentifierType.TABLE) {
+      graphqlName = escapeReservedPrefix(graphqlName);
+      graphqlName = escapeReservedSuffix(graphqlName);
+    } else if (type == IdentifierType.UDT) {
+      graphqlName = escapeReservedPrefix(graphqlName);
+      // No need to escape suffixes because we append our own:
+      graphqlName += "Udt";
+    }
+    return graphqlName;
+  }
+
+  public static String toCql(String graphqlName, IdentifierType type) {
+    if (graphqlName == null || graphqlName.isEmpty()) {
+      throw new IllegalArgumentException("GraphQL name must be non-null and not empty");
+    }
+    if (type == IdentifierType.UDT) {
```

Comment:
minor: From Codacy this could be: `if (type == IdentifierType.UDT && graphqlName.endsWith("Udt")) {`